conn/physic/units_test.go

```diff
@@ -536,3 +537,1260 @@ func BenchmarkCelsiusFloatg(b *testing.B) {
+		{"1SiUnitForce", &newton, "1N", false, ""},
+		{"1SiUnitFrequency", &hertz, "1Hz", false, ""},
+		{"1SiUnitMass", &gram, "1g", false, ""},
+		{"1SiUnitPressure", &pascal, "1Pa", false, ""},
+		{"1SiUnitRelativeHumidity", &humidity, "1%rH", false, ""},
+		{"1SiUnitSpeed", &metresPerSecond, "1m/s", false, ""},
+		{"1SiUnitTemperature", &celsius, "1K", false, ""},
+		{"1SiUnitPower", &watt, "1W", false, ""},
+		{"1SiUnitEnergy", &joule, "1J", false, ""},
+		{"1SiUnitLuminousIntensity", &candela, "1cd", false, ""},
+		{"1SiUnitLuminousFlux", &lux, "1lm", false, ""},
+		// Naked values
+		{
+			"noUnitErrAngle", &degree, "1", true,
+			"parse error: no units provided, need: \"Radian\"",
+		},
+		{
+			"noUnitErrDistance", &metre, "1", true,
+			"parse error: no units provided, need: \"m\"",
+		},
+		{
+			"noUnitErrElectricCurrent", &amp, "1", true,
+			"parse error: no units provided, need: \"A\"",
+		},
+		{
+			"noUnitErrElectricPotential", &volt, "1", true,
+			"parse error: no units provided, need: \"V\"",
+		},
+		{
+			"noUnitErrElectricResistance", &ohm, "1", true,
+			"parse error: no units provided, need: \"Ohm\"",
+		},
+		{
+			"noUnitErrElectricalCapacitance", &farad, "1", true,
+			"parse error: no units provided, need: \"F\"",
+		},
+		{
+			"noUnitErrForce", &newton, "1", true,
+			"parse error: no units provided, need: \"N\"",
+		},
+		{
+			"noUnitErrFrequency", &hertz, "1", true,
+			"parse error: no units provided, need: \"Hz\"",
+		},
+		{
+			"noUnitErrMass", &gram, "1", true,
+			"parse error: no units provided, need: \"g\"",
+		},
+		{
+			"noUnitErrPressure", &pascal, "1", true,
+			"parse error: no units provided, need: \"Pa\"",
+		},
+		{
+			"noUnitErrRelativeHumidity", &humidity, "1", true,
+			"parse error: no units provided, need: \"%rH\"",
+		},
+		{
+			"noUnitErrSpeed", &metresPerSecond, "1", true,
+			"parse error: no units provided, need: \"m/s\"",
+		},
+		{
+			"noUnitErrTemperature", &celsius, "1", true,
+			"parse error: no units provided, need: \"K or C\"",
+		},
+		{
+			"noUnitErrPower", &watt, "1", true,
+			"parse error: no units provided, need: \"W\"",
+		},
+		{
+			"noUnitErrEnergy", &joule, "1", true,
+			"parse error: no units provided, need: \"J\"",
+		},
+		{
+			"noUnitErrLuminousIntensity", &candela, "1", true,
+			"parse error: no units provided, need: \"cd\"",
+		},
+		{
+			"noUnitErrLuminousFlux", &lux, "1", true,
+			"parse error: no units provided, need: \"lm\"",
+		},
+		{
+			"overflow", &joule, "123456789012345678901234567890", true,
+			"parse error: overflows maximum is: \"9223372036854775807\"",
+		},
+	}
+
+	for _, tt := range tests {
+		got := tt.v.Set(tt.s)
+		if tt.wantErr && got == nil {
+			t.Errorf("case %s expected error but got none", tt.name)
+		}
+		if !tt.wantErr && got != nil {
+			t.Errorf("case %s got %v", tt.name, got)
+		}
+		if tt.wantErr && got.Error() != tt.err {
+			t.Errorf("case %s got %v", tt.name, got)
+		}
+	}
+}
+
+func TestAtod(t *testing.T) {
```

Comment:
I'd like a first PR that only adds atod and dtoi without user visible changes but with the unit tests, then another change with the user visible tests. That would make reviewing easier. These functions are quite tricky.